llvm/utils/lit/lit/builtin_commands/cat.py

```diff
@@ -69,8 +87,8 @@ def main(argv):
                 fileToCat = open(filename, "rb")
                 contents = fileToCat.read()
 
-            if show_nonprinting:
-                contents = convertToCaretAndMNotation(contents)
+            if True in fields(enabled_options):
+                contents = convertTextNotation(contents, enabled_options)
```

Comment:
What happens if I add a field to `Options` that isn't either `show_nonprinting` or `show_ends`?

Reply:
If both `show_nonprinting` and `show_ends` are False and this added field was True, `convertTextNotation` has an `else` statement on lines 43/44 that should account for this case, and not change the text (assuming that there is no existing implementation for this new option). But this is just a precaution, because I don't think this case should technically ever occur though - any option that exists in `Options` should have an implementation in `convertTextNotation`. Does this address your concern or am I misunderstanding?

Reply:
But that means well run the conversion code when we don't need to right? It's preferable in my opinion to avoid that work if we don't have to, specially since that routine a character by character copy.

I'm also wondering if we should have an early exit from the routine if the neither option is set. WDYT?